std/traits.d

```diff
@@ -2658,6 +2659,150 @@ template RepresentationTypeTuple(T)
     static assert(R2.length == 2 && is(R2[0] == int) && is(R2[1] == immutable(Object)));
 }
 
+// Layout
+/**
+Returns the layout in memory of type `T` as an `AliasSeq`. For each field in
+`T`, the result includes the offset of the field followed by the type of the
+field. Detailed information depends on the type's kind, as follows:
+
+$(UL
+$(LI If `T` is a `struct` or `union` type, then `Layout!T` is an `AliasSeq`
+containing each field preceded by its offset. Fields that are in turn of
+`struct` or `union` types are decomposed transitively. The result is ordered
+in increasing offset order. (Two or more fields may have the same offset due to
+`union`s.) For a given offset, types are in the order of the declaration
+they are a part of.)
+
+$(LI If `T` is a `class` or `interface` type, the layout is similar to the one
+return for `struct`s, with one difference: classes and interfaces have
+additional opaque data managed by the compiler. Usually these are placed at the
+front of the object. This data is represented in the layout as pointers to opaque
+types. The presence, number, and types of additional fields are
+implementation-dependent and should not be relied upon in portable code. All these
+additional types have names that start with a double underscore.)
+
+$(LI If `T` is a statically-sized array, the layout is an iteration of the
+layouts of each field of the array. Note that this may create large tuples.)
+
+$(LI For all other types, the layout is $(D AliasSeq!(size_t(0), T)).)
```

Comment:
The documentation enumerates cases 1,2,3,4. However, the code is laid out as cases 3,4,(1,2). The code can be laid out in the same order as the documentation. Code comments should be added to say which case # they are dealing with.